pkg/operation/seed/seed.go

```diff
@@ -685,9 +694,32 @@ func BootstrapCluster(ctx context.Context, k8sGardenClient, k8sSeedClient kubern
 		}
 	}
 
+	if managedIngress(seed) {
+		seedClient := k8sSeedClient.Client()
+
+		// TODO(beckermax): Remove after next release - temporary migration logic.
+		if err := migrateIngressClassForShootIngresses(ctx, seedClient); err != nil {
+			return err
+		}
+
+		dnsProvider := seed.Info.Spec.DNS.Provider
+		if dnsProvider == nil {
+			return fmt.Errorf("managed Ingress is requested but dns Provider is missing configuration")
+		}
```

Comment:
No strong opinion here, but this is/should already been validated in the API validation.